cookbook/bundles/tests.md

```diff
@@ -0,0 +1,255 @@
+.. index::
+   single: Bundle; Testing third party bundles
+
+Writing tests for third party bundles
+=====================================
+
+Let's have a look at a minimized version of the Standard Edition's front controller:
+
+    <?php
+    // File: web/app.php
+
+    use Symfony\Component\HttpFoundation\Request;
+
+    require_once __DIR__.'/../vendor/autoload.php';
+    require_once __DIR__.'/../app/AppKernel.php';
+
+    // Create the request from PHP variables super globals such as $_GET, $_POST, $_SERVER, etc
+    $request = Request::createFromGlobals();
+
+    // The application handles the Request
+    $kernel = new AppKernel('prod', false);
+    $response = $kernel->handle($request);
+
+    // Dump the Response's headers and print the Response's content
+    $response->send();
+
+In this example Symfony is used as an application that mimicks the HTTP protocol,
+which means that it is possible to write transversal tests (System Tests) by building
+"manually" a ``Request``, giving it to our application and then check the returned ``Response``.
+
+This article provides examples on how to write such tests with third party bundles.
+
+Creating a minimal AppKernel
+----------------------------
+
+Third party bundles cannot be run on their own, they need an application. Instead
+of creating a separate application, install the bundle in it and then check how things turn out,
+it is possible to create a minimalistic ``AppKernel`` inside the bundle, for testing
+and showcase purposes.
+
+If the third party bundle is named ``AcmeMyBundle``, the ``AppKernel`` could look like this:
+
+    <?php
+    // File: Tests/app/AppKernel.php
+
+    namespace Acme\MyBundle\Tests;
+
+    use Symfony\Component\HttpKernel\Kernel;
+    use Symfony\Component\Config\Loader\LoaderInterface;
+
+    class AppKernel extends Kernel
+    {
+        public function registerBundles()
+        {
+            return array(
+                new Symfony\Bundle\FrameworkBundle\FrameworkBundle(),
+                new Acme\MyBundle\AcmeMyBundle(),
+            );
+        }
+
+        public function registerContainerConfiguration(LoaderInterface $loader)
+        {
+            $loader->load(__DIR__.'/config.yml');
+        }
+    }
+
+.. note::
+
+    ``AppKernel`` adds a direct dependency on Symfony's FrameworkBundle, which can be
+    installed by running the following command: ``composer require --dev symfony/framework-bundle``.
+
+In its minimum state ``AppKernel`` requires only one configuration parameter, ``secret``:
+
+    # File: Tests/app/config.yml
```

Comment:
as it's not PHP code, you have to use the long version here (`.. code-block:: yaml`) + one empty line before this line and this line should be `# Tests/app/config.yml`